prog/vnet/subnet_nexus.rb

```diff
@@ -20,11 +23,19 @@ def self.assemble(project_id, name: nil, location: "hetzner-hel1", ipv6_range: n
     DB.transaction do
       ps = PrivateSubnet.create(name: name, location: location, net6: ipv6_range, net4: ipv4_range, state: "waiting") { _1.id = ubid.to_uuid }
       ps.associate_with_project(project)
-      port_range = allow_only_ssh ? 22..22 : 0..65535
-      fw = Firewall.create_with_id(name: "#{name}-default")
-      fw.associate_with_project(project)
-      ["0.0.0.0/0", "::/0"].each { |cidr| FirewallRule.create_with_id(firewall_id: fw.id, cidr: cidr, port_range: Sequel.pg_range(port_range)) }
-      fw.associate_with_private_subnet(ps, apply_firewalls: false)
+
+      firewall = if firewall_id
+        existing_fw = project.firewalls_dataset.first(Sequel[:firewall][:id] => firewall_id)
+        fail "Firewall with id #{firewall_id} does not exist" unless existing_fw
+        existing_fw
+      else
+        port_range = allow_only_ssh ? 22..22 : 0..65535
+        new_fw = Firewall.create_with_id(name: "#{name}-default")
+        new_fw.associate_with_project(project)
+        ["0.0.0.0/0", "::/0"].each { |cidr| FirewallRule.create_with_id(firewall_id: new_fw.id, cidr: cidr, port_range: Sequel.pg_range(port_range)) }
+        new_fw
+      end
+      firewall.associate_with_private_subnet(ps, apply_firewalls: false)
```

Comment:
Interesting way to keep the default. I appreciate we can defer what to do about this sticky matter.